modules/library/metadata/src/main/java/org/geotools/util/logging/LoggingImagingListener.java

```diff
@@ -29,10 +30,15 @@ public boolean errorOccurred(
         Logger log = Logging.getLogger(LoggingImagingListener.class);
         if (message.contains("Continuing in pure Java mode")) {
             log.log(Level.FINER, message, thrown);
+            return false; // we are not trying to recover
         } else {
             log.log(Level.INFO, message, thrown);
+            if (thrown instanceof RuntimeException && !(where instanceof OperationRegistry)) {
+                throw (RuntimeException) thrown;
```

Comment:
We should be clear that this represents a change in functionality for the GeoTools library, I hope it does not cause too much trouble!